src/twisted/runner/procmon.py

```diff
@@ -164,15 +233,15 @@ def removeProcess(self, name):
         @param name: A string that uniquely identifies the process.
         """
         self.stopProcess(name)
-        del self.processes[name]
+        del self._processes[name]
 
 
     def startService(self):
         """
         Start all monitored processes.
         """
         service.Service.startService(self)
-        for name in self.processes:
+        for name in self._processes:
```

Comment:
Should this (and stopService) be reentrant? I think that you may need to do `for name in list(self._processes)`.